File: personal_page_backend/src/main/java/hu/kyberszittya/personal/backend/model/BlogPost.java
package hu.kyberszittya.personal.backend.model;

import jakarta.persistence.*;

import java.util.Date;


@Entity
public class BlogPost {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private Date date;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    /// Getters
    ///
    ///

    public String getTitle() {
        return title;
    }

    public Date getDate() {
        return date;
    }

    public String getContent() {
        return content;
    }

}

File: personal-page/src/models/BlogPost.ts
export class BlogPost {
    id: number;
    title: string;
    content: string;
    date: string;
  
    constructor(id: number, title: string, content: string, date: string) {
      this.id = id;
      this.title = title;
      this.content = content;
      // If the date is passed as a string (from API), convert it to a Date object
      this.date = date;
    }
  
    
  }